src/modules/facility-fixed-fee/facility-fixed-fee.service.test.ts

```diff
@@ -206,5 +222,74 @@ describe('FacilityFixedFeeService', () => {
     });
   });
 
+  describe('createAmountAmendmentForFixedFees', () => {
+    const facilityIdentifier = valueGenerator.facilityId();
+    const createdBundleIdentifier = valueGenerator.acbsBundleId();
+    const acbsBundleCreatedResponse: AcbsCreateBundleInformationResponseHeadersDto = { BundleIdentifier: createdBundleIdentifier };
+    const { facilityFeeTransactionType } = PROPERTIES.FACILITY_FEE_AMOUNT_TRANSACTION.DEFAULT.bundleMessageList;
+
+    const { increaseAmountRequest, decreaseAmountRequest, acbsFixedFeesAmendmentForIncrease, acbsFixedFeesAmendmentForDecrease } =
+      new CreateFacilityFixedFeesAmountAmendmentGenerator(valueGenerator, dateStringTransformations).generate({ numberToGenerate: 3, facilityIdentifier });
+
+    describe('when creating a fixed fees amendment bundle in ACBS that increases the amount', () => {
+      beforeEach(() => {
+        when(createBundleInformation).calledWith(acbsFixedFeesAmendmentForIncrease, idToken).mockResolvedValueOnce(acbsBundleCreatedResponse);
+      });
+
+      it('returns the bundleIdentifier from creating the fixed fees amendment bundle', async () => {
+        const response = await service.createAmountAmendmentForFixedFees(facilityIdentifier, increaseAmountRequest);
+
+        expect(response.bundleIdentifier).toBe(createdBundleIdentifier);
+      });
+
+      it('uses the increase FacilityFeeTransactionType when creating the fixed fees amendment bundle', async () => {
+        await service.createAmountAmendmentForFixedFees(facilityIdentifier, increaseAmountRequest);
+
+        const createdBundleInAcbs = getBundleCreatedInAcbs();
+
+        expect(createdBundleInAcbs.BundleMessageList[0].FacilityFeeTransactionType.TypeCode).toBe(facilityFeeTransactionType.increaseTypeCode);
+      });
+
+      it('sets the amountAmendment as the TransactionAmount when creating the fixed fees amendment bundle', async () => {
+        await service.createAmountAmendmentForFixedFees(facilityIdentifier, increaseAmountRequest);
+
+        const createdBundleInAcbs = getBundleCreatedInAcbs();
+
+        expect(createdBundleInAcbs.BundleMessageList[0].TransactionAmount).toBe(increaseAmountRequest[0].amountAmendment);
+      });
+    });
+
+    describe('when creating a fixed fees amendment bundle in ACBS that decreases the amount', () => {
```

Comment:
Can we add an assertion where the amount is `0` please.

Reply:
Client input validation is tested in api-test. Api-test output has this error check "amountAmendment should not be equal to 0"

    withNumberFieldValidationApiTests({
      fieldName: 'amountAmendment',
      validRequestBody: increaseAmountRequest,
      makeRequest,
      givenAnyRequestBodyWouldSucceed,
      forbidZero: true,
    });